### PR TITLE
http-01 wellknown support + ansible galaxy file structure :

### PR DESCRIPTION
**Pull request by DnR-iData**
+ http-01 wellknown support
+ dynamic ssl url var set
+ haproxy crt + key concatenation
+ ansible galaxy file structure

### CODE REVIEW

**Comment by andrewhowdencom** [comment]
Holy shit this is awesome <3 I will review now.

File: meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  # Some suggested licenses:
+  # - BSD (default)
+  # - MIT
+  # - GPLv2
+  # - GPLv3
+  # - Apache
+  # - CC-BY
+  license: license (GPLv2, CC-BY, etc)
+
+  min_ansible_version: 1.2
```
Comment (andrewhowdencom) [inline meta/main.yml]: This requires some RSA stuff only available in 2.4+

https://github.com/sitewards/ansible-role-lets-encrypt/blob/master/tasks/lets_encrypt.yml#L43
http://docs.ansible.com/ansible/latest/openssl_csr_module.html

File: tasks/main.yml [hunk 2]
```diff
@@ -1,4 +1,28 @@
 ---
 - include: "dependencies.yml"
+
+# Create the /.well-known/acme-challenge dir
+- include: "wellknow.yml"
```
Comment (andrewhowdencom) [inline tasks/main.yml]: this looks to be spelled incorrectly -- i think it's supposed to be `wellknown.yml`

File: tasks/providers/http-01/wellknown.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+#      # acme_data.challenge_data =>
```
Comment (andrewhowdencom) [inline tasks/providers/http-01/wellknown.yml]: What is the purpose of this comment? If I had to hazard a guess, I would imagine it was left here during debugging?

File: tasks/main.yml (hunk 2 above)
Comment (andrewhowdencom) [reply tasks/main.yml]: Ah, I am incorrect - I see this now. Still, I think it makes more sense if they're both `wellknown` -- what do you think?

**Comment by andrewhowdencom** [comment]
This is excellent, and the work is much appreciated. I have done a quick review, and a couple of things stand out. While I'd be happy to merge it as is (it does improve things), the marks on review would be thing I'd love to see tidied up before it hits main. 

It's super appreciated. <3

**Comment by DnR-iData** [comment]
Hi,

Happy to see the feature is appreciated :)

I have done the corrections (typo, version).

Thank you 👍

**Comment by aboritskiy** [comment]
Hi, it is appreciated, thank you for the contribution!

**Comment by andrewhowdencom** [comment]
Heya @DnR-iData; 

I'm super sorry it took this long to get back to you. Unfortunately this was mine to Shepard into merging, and I simply lost it in the notification spam. 

It's merged now. Thanks again for your patience <3

**Pull request merged**